### PR TITLE
Update 'ZLIB' to v1.2.11-p0

### PR DESCRIPTION
**Pull request by hjmallon**
* I've followed [this guide](https://docs.hunter.sh/en/latest/creating-new/update.html)
  step by step carefully. **[Yes]**

* I've tested this package remotely and have excluded all broken builds.
  Here is the links to the Travis/AppVeyor with status "All passed":

  * https://ci.appveyor.com/project/hjmallon/hunter/builds/22692920
  * https://travis-ci.org/hjmallon/hunter/builds/499331972

### CODE REVIEW

File: cmake/configs/default.cmake [hunk 1]
```diff
@@ -176,7 +176,7 @@ hunter_default_version(autobahn-cpp VERSION 0.2.0)
 hunter_default_version(autoutils VERSION 0.3.0)
 hunter_default_version(aws-c-common VERSION 0.2.1-p0)
 hunter_default_version(benchmark VERSION 1.4.0)
-hunter_default_version(binaryen VERSION 1.38.28-p1)
```
Comment (ruslo) [inline cmake/configs/default.cmake]: Please send the changes that are not related to `ZLIB` as a separate pull request.

File: cmake/projects/ZLIB/hunter.cmake [hunk 2]
```diff
@@ -34,29 +45,28 @@ hunter_add_version(
...
-    "https://github.com/hunter-packages/zlib/archive/v1.2.8-hunter.tar.gz"
+    "https://github.com/hunter-packages/zlib/archive/v1.2.11-p0.tar.gz"
     SHA1
-    75a05fcc928ed52e1eeb93f07a1c78a7890860c0
+    9ced8b310aa341af7f3584bac0b2d2a669d2178b
 )
 
 hunter_pick_scheme(DEFAULT url_sha1_cmake)
 hunter_cacheable(ZLIB)
-hunter_download(PACKAGE_NAME ZLIB
```
Comment (ruslo) [inline cmake/projects/ZLIB/hunter.cmake]: Please send code cleanups as a separate pull request.

File: cmake/configs/default.cmake (hunk 1 above)
Comment (hjmallon) [reply cmake/configs/default.cmake]: Sorry, I will turn off my whitespace cleaner.

Comment (ruslo) [reply cmake/configs/default.cmake]: > Sorry, I will turn off my whitespace cleaner.

No, it's fine, just send it separately.

**Comment by hjmallon** [comment]
I have closed this to split up the changes
